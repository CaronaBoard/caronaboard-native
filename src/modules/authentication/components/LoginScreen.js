import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkTextInput, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton } from '../../shared/components/index'
import { styles } from '../containers/styles/LoginScreenStyles'

export const INITIAL_STATE = {
  email: '',
  password: '',
  loading: false
}

export default class LoginV2 extends React.Component {
  constructor (props) {
    super(props)
    this.state = INITIAL_STATE
  }

  onButtonPress = () => {
    const { email, password } = this.state
    const { onButtonPress } = this.props
    try {
      onButtonPress(email, password)
    } catch (error) {
      console.log(error)
    }
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.alert

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/logo.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <RkTextInput onChangeText={(email) => { this.setState({email}) }} rkType='rounded' placeholder='Username' autoCapitalize='none' />
        <RkTextInput onChangeText={(password) => { this.setState({password}) }} rkType='rounded' placeholder='Password' secureTextEntry />

        <View style={styles.centralized}>
          <GradientButton
            style={styles.loginButton}
            rkType='large'
            text='LOGIN'
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer()}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
