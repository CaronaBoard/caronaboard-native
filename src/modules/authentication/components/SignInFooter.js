import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import { RkButton, RkText } from 'react-native-ui-kitten'
import Styles from './styles/SignInFooterStyles'
import { screens } from '../../../configuration/navigation/Screens'

export class SignInFooter extends Component {
  static propTypes = {
    email: PropTypes.string,
    navigator: PropTypes.object.isRequired
  }

  navigateTo = screen => () => this.props.navigator.push({screen})

  render () {
    return (
      <View>
        <View style={Styles.textRow}>
          <RkButton
            key='forgot-password-button'
            rkType='clear'
            onPress={this.navigateTo(screens.forgotPassword.id)}
          >
            <RkText rkType='header6'>Forgot your password?</RkText>
          </RkButton>
        </View>
        <View style={Styles.textRow}>
          <RkText rkType='primary3'>Donâ€™t have an account?</RkText>
          <RkButton
            key='signup-button'
            rkType='clear'
            onPress={this.navigateTo(screens.signUp.id)}
          >
            <RkText rkType='header6'> Sign up now</RkText>
          </RkButton>
        </View>
      </View>
    )
  }
}
