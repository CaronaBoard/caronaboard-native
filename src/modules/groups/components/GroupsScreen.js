import React from 'react'
import { RkText } from 'react-native-ui-kitten'
import {
  ScrollView,
  View,
  TouchableOpacity
} from 'react-native'
import PropTypes from 'prop-types'

import { styles } from './styles/GroupsScreen.style'
import { GroupPropType } from '../types'
import { WarningScreen } from '../../shared/components/WarningScreen.style'

const hardCodedGroup = {
  'id': 'twpoa',
  'members': {},
  'name': 'ThoughtWorks POA'
}

export class GroupsScreen extends React.Component {
  static propTypes = {
    uid: PropTypes.string.isRequired,
    groups: PropTypes.arrayOf(GroupPropType).isRequired,
    joinGroup: PropTypes.func.isRequired,
    changeGroup: PropTypes.func.isRequired,
    activeGroup: PropTypes.oneOfType([GroupPropType, {}])
  }

  renderActiveGroup = (title, activeGroup) => {
    if (!activeGroup.name) {
      return null
    }

    return (
      <View style={styles.section}>
        <View style={[styles.row, styles.heading]}>
          <RkText rkType='primary header6'>{title}</RkText>
        </View>
        <View style={styles.row} >
          <View
            style={styles.rowButton}
          >
            <RkText rkType='header6'>{activeGroup.name}</RkText>
          </View>
        </View>
      </View>
    )
  }

  renderGroupList = (title, group, onPress = () => null) => {
    return (
      <View style={styles.section}>
        <View style={[styles.row, styles.heading]}>
          <RkText rkType='primary header6'>{title}</RkText>
        </View>
        {group.map(group => {
          return (
            <View style={styles.row} key={group.id}>
              <TouchableOpacity
                style={styles.rowButton}
                onPress={() => onPress(group)}
              >
                <RkText rkType='header6'>{group.name}</RkText>
              </TouchableOpacity>
            </View>
          )
        })}
      </View>
    )
  }

  render () {
    const {groups, uid, joinGroup, changeGroup, activeGroup} = this.props
    const groupsYouBelong = [hardCodedGroup]

    groups.forEach(group => {
      if (group.members.hasOwnProperty(uid)) {
        groupsYouBelong.push(group)
      }
    })

    if (!uid) {
      return (
        <WarningScreen activeGroup={'Must have profile first'} />
      )
    }

    return (
      <ScrollView style={styles.container}>
        {this.renderActiveGroup('GRUPO ATIVO', activeGroup)}
        {this.renderGroupList('GRUPOS QUE VOCE PERTENCE', groupsYouBelong, changeGroup)}
        {this.renderGroupList('TODOS OS GRUPOS', groups, joinGroup)}
      </ScrollView>
    )
  }
}
