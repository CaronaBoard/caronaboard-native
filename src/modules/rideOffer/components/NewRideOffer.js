import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { RkText } from 'react-native-ui-kitten'

import styles from './styles/YourRideOffer.style'
import { TextInput } from '../../shared/components/index'
import { LoadingSpinnerView } from '../../shared/components/LoadingSpinnerView'
import { GradientButton } from '../../shared/components/GradientButton'

export const INITIAL_STATE = {
  origin: '',
  destination: '',
  days: '',
  hours: '',
  loading: false
}

export class NewRideOffer extends React.PureComponent {
  state = INITIAL_STATE

  static propTypes = {
    onPress: PropTypes.func.isRequired
  }

  onPress = async () => {
    this.setState({loading: true})
    const { origin, destination, days, hours } = this.state
    await this.props.onPress({ origin, destination, days, hours })
    this.setState({loading: false})
  }

  render () {
    return (
      <LoadingSpinnerView isLoading={this.state.loading} style={styles.flexible}>
        <View style={styles.container}>
          <RkText style={styles.title}>
            Ride Offer
          </RkText>
          <View style={styles.inputTextsContainer}>
            <TextInput placeholder='ORIGEM' onChangeText={(origin) => this.setState({origin})} />
            <TextInput placeholder='DESTINO' onChangeText={(destination) => this.setState({destination})} />
            <TextInput placeholder='DIAS' onChangeText={(days) => this.setState({days})} />
            <TextInput placeholder='HORAS' onChangeText={(hours) => this.setState({hours})} />
            <View style={styles.centralized}>
              <GradientButton
                rkType='large'
                text={'Oferecer carona'}
                onPress={this.onPress}
              />
            </View>
          </View>
        </View>
      </LoadingSpinnerView>
    )
  }
}
