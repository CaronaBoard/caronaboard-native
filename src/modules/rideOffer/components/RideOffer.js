import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'
import Styles from './styles/RideOfferStyles'
import Icon from 'react-native-vector-icons/MaterialIcons'
import { RkCard } from 'react-native-ui-kitten'
import { ridePropTypes } from '../../rideRequest/types'

const renderLine = (text, icon) => {
  return (
    <View style={Styles.line} >
      <Icon name={icon} style={Styles.icon} />
      <Text rkCardText>{text}</Text>
    </View>
  )
}

export const RideOffer = props => {
  const {
    origin,
    destination,
    days,
    hours
  } = props.ride

  return (
    <TouchableOpacity>
      <RkCard>
        <View rkCardContent>
          { renderLine(origin, 'radio-button-unchecked') }
          { renderLine(destination, 'radio-button-unchecked') }
        </View>
        <View rkCardContent>
          { renderLine(days, 'today') }
          { renderLine(hours, 'schedule') }
        </View>
      </RkCard>
    </TouchableOpacity>
  )
}

RideOffer.propTypes = {
  ride: ridePropTypes
}
