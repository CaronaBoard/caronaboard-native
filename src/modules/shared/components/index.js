export { Button } from './Button'
export { TextInput } from './TextInput'
export { GradientButton } from './GradientButton'
export { LinearGradient } from './LinearGradient'
