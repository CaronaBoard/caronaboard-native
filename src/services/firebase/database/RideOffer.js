// @flow

import Firebase from 'firebase'
import { toArrayOfRides } from '../Conversion'
import _ from 'lodash'
import type { rideType, rideOfferType } from '../types'
import { rideGroup } from './'
import { getUserProfile } from './Profile'
import type { profileFlowType } from './Profile'

export const saveRideOffer = async (rideOffer: rideOfferType, userId: string) => {
  const profile = await getUserProfile(userId)

  return Firebase.database()
    .ref(`rides/${rideGroup}/${userId}`)
    .push(Object.assign({ profile }, rideOffer))
}

export const updateRideOffer = async (rideOffer: rideOfferType, userId: string, profile: profileFlowType) => {
  await Firebase.database()
    .ref(`rides/${rideGroup}/${userId}/${rideOffer.id}`)
    .update(Object.assign({profile: profile}, _.omit(rideOffer, 'id')))
}

export const removeRideOffer = async (rideId: string, userId: string) => {
  return Firebase.database()
    .ref(`rides/${rideGroup}/${userId}/${rideId}`)
    .remove()
}

export const getAllRideOffers = async (): Array<rideType> => {
  const rides = await Firebase.database()
    .ref('rides')
    .child(rideGroup)
    .once('value')

  return toArrayOfRides(rides.val())
}

export const getUserRideOffers = async (userId: string) => {
  const rides = await getAllRideOffers()
  return rides.filter(ride => ride.driverId === userId)
}
