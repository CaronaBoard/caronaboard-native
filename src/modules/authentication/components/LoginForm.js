import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton, TextInput } from '../../shared/components'
import { styles } from '../containers/styles/LoginScreenStyles'

export class LoginForm extends React.Component {
  static defaultProps = {
    footer: () => {},
    toast: {
      showAlert: false,
      message: ''
    }
  }

  state = {
    email: '',
    password: '',
    loading: false
  }

  onButtonPress = () => {
    const { email, password } = this.state
    this.props.onButtonPress(email, password)
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.toast

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/caronaboard-logo-azul.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <TextInput
          onChangeText={(email) => { this.setState({email}) }}
          key='email-input'
          rkType='rounded'
          placeholder='Username'
          autoCapitalize='none'
        />
        <TextInput
          onChangeText={(password) => { this.setState({password}) }}
          key='password-input'
          rkType='rounded'
          placeholder='Password'
          secureTextEntry
        />

        <View style={styles.centralized}>
          <GradientButton
            rkType='large'
            style={styles.loginButton}
            text={this.props.buttonText}
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer(this.state.email)}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
