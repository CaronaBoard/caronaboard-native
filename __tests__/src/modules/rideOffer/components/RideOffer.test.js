import React from 'react'
import { shallow } from 'enzyme'
import { RideOffer } from '../../../../../src/modules/rideOffer/components/RideOffer'

const props = {
  ride: {}
}

describe('<RideOffer />', () => {
  const wrapper = shallow(<RideOffer {...props} />)

  it('Should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
