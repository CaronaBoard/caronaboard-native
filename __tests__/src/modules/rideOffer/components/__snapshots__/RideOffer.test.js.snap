// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RideOffer /> Should have a snapshot 1`] = `
<LoadingSpinnerView
  isLoading={false}
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
  >
    <RkCard>
      <View
        rkCardContent={true}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="radio-button-unchecked"
            size={12}
            style={
              Object {
                "fontSize": 15,
                "marginRight": 5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            rkCardText={true}
          >
            PUC
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="radio-button-unchecked"
            size={12}
            style={
              Object {
                "fontSize": 15,
                "marginRight": 5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            rkCardText={true}
          >
            Bomfim
          </Text>
        </View>
      </View>
      <View
        rkCardContent={true}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="today"
            size={12}
            style={
              Object {
                "fontSize": 15,
                "marginRight": 5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            rkCardText={true}
          >
            Seg-Sex
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="schedule"
            size={12}
            style={
              Object {
                "fontSize": 15,
                "marginRight": 5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            rkCardText={true}
          >
            19h
          </Text>
        </View>
      </View>
      <Icon
        allowFontScaling={false}
        color="#900"
        name="delete-forever"
        size={30}
        style={
          Object {
            "alignSelf": "flex-end",
            "margin": 10,
          }
        }
      />
    </RkCard>
  </TouchableOpacity>
</LoadingSpinnerView>
`;
